clients/src/main/java/org/apache/kafka/common/utils/ByteBufferOutputStream.java

```diff
@@ -105,7 +105,7 @@ public int initialCapacity() {
         return initialCapacity;
     }
 
-    private void maybeExpandBuffer(int remainingRequired) {
+    public void maybeExpandBuffer(int remainingRequired) {
```

Comment:
Hmm.. I debated it, but ultimately decided against it so that `writeEmptyHeader` was somewhat consistent with `writeHeader`. Also, exposing `maybeExpandBuffer` didn't seem like a bad idea anyway.